WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetViewModel.kt

```diff
@@ -63,15 +69,55 @@ class ViewsWidgetViewModel
         }
     }
 
+    fun loadSites() {
+        mutableSites.postValue(siteStore.sites.map { it.toUiModel() })
+    }
+
+    private fun SiteModel.toUiModel(): SiteUiModel {
+        val blogName = SiteUtils.getSiteNameOrHomeURL(this)
+        val homeUrl = SiteUtils.getHomeURLOrHostName(this)
+        val title = when {
+            !blogName.isNullOrEmpty() -> blogName
+            !homeUrl.isNullOrEmpty() -> homeUrl
+            else -> null
+        }
+        val description = when {
+            !homeUrl.isNullOrEmpty() -> homeUrl
+            else -> null
+        }
+        return SiteUiModel(this.siteId, this.iconUrl, title, description) {
+            selectSite(it)
+        }
+    }
+
+    private fun selectSite(site: SiteUiModel) {
+        if (mutableSelectedSite.value != site) {
```

Comment:
The problem is that the 2 objects are not equal because both have a different anonymous `onClick` function. I've fixed that but thinking about it now I don't think it really makes sense. We want the user to be able to reselect a site so I'll remove the `if`